plugins/x86/x86_prefix.mli

```diff
@@ -0,0 +1,22 @@
+open Core_kernel.Std
+open Bap.Std
+       
+type t = [
+  | `_F0 (** LOCK                *)
+  | `_F2 (** REPNE               *)
+  | `_F3 (** REP/REPE            *)
+  | `_2E (** CS segment override *)
+  | `_36 (** SS segment override *)
+  | `_3E (** DS segment override *)
+  | `_26 (** ES segment override *)
+  | `_64 (** FS segment override *)
+  | `_65 (** GS segment override *)
+  | `_66 (** operand override    *)
+  | `_67 (** address override    *)
+] [@@deriving bin_io, sexp, compare, enumerate]
+
+(** [get mem] - returns a list of prefixes, if any *)
+val get : mem -> t list Or_error.t
+
+(** [get_opt mem] - returns a list of prefixes, if any *)
+val get_opt : mem -> t list option
```

Comment:
This sound like a general excuse, that totally doesn't suite this case. What kind of reason you're trying to emphasize? Look at the code, the only error condition that you raise is called `"opcode not found"`. How does it differ from the `None`? You do not add any extra semantics. Moreover, even this error condition is subsumed in the `get_opt` function, so you can actually distinguish between the error condition, and incorrect prefix. So no, I'm not taking this excuse. The function type doesn't have any sense.